jmclient/jmclient/configure.py

```diff
@@ -749,6 +753,34 @@ def load_program_config(config_path="", bs=None, plugin_services=[]):
                 os.makedirs(plogsdir)
             p.set_log_dir(plogsdir)
 
+def gracefully_kill_subprocess(p):
+    # See https://stackoverflow.com/questions/43274476/is-there-a-way-to-check-if-a-subprocess-is-still-running
+    if p.poll() is None:
+        p.send_signal(SIGINT)
+
+def check_and_start_tor():
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    result = sock.connect_ex(("127.0.0.1", 9050))
+    sock.close()
+    if result == 0:
+        return
+    log.info("Nobody listens on 127.0.0.1:9050, trying to start Tor.")
```

Comment:
I see we try, if you have some other instance or system tor daemon running, to use it. But if it could be unavailable anyway, Tor configuration support the port value `auto` picking a random user available >1024 unused port at startup. But JM would need to learn it.

Alternatively, using unix:sockets for this communication is possible too.

Reply:
If user has manually changed `torrc` from the defaults, he should not use this feature and just stick to existing Tor instance by changing port settings in `joinmarket.cfg`.